setup.py

```diff
@@ -390,7 +401,38 @@ def run(self) -> None:
             ext for ext in self.extensions
             if not isinstance(ext, CMakeExtension)
         ]
+
+        if "paddle" in frameworks():
+            # Add linker search path for libtransformer_engine.so
+            self.extensions[0].library_dirs.append(self.build_lib)
+
         super().run()
+
+        # Paddle needs to manually write stub
+        if "paddle" in frameworks():
+            def find_transformer_engine(path):
+                """Finds the path for libtransformer_engine"""
+                for filename in os.listdir(path):
+                    if filename.startswith("libtransformer_engine"):
+                        return os.path.join(path, filename)
+
+            from paddle.utils.cpp_extension.extension_utils import custom_write_stub
+            module_name = self.extensions[0].name
+            assert module_name[-4:] == '_pd_', "Expect module name end with '_pd_'"
+            stub_name = module_name[:-4] # remove '_pd_'
+            so_path = self.get_ext_fullpath(module_name)
+            _, ext = os.path.splitext(so_path)
+            pyfile_path = os.path.join(
+                    self.build_lib, stub_name + '.py')
+            print("Writing paddle stub for {} into file {}".format(
+                    stub_name + ext, pyfile_path))
+            # Fixes undefined symbol error for paddle stub writing
+            import ctypes
+            ctypes.CDLL(find_transformer_engine(self.build_lib),
+                        mode=ctypes.RTLD_GLOBAL)
+
+            custom_write_stub(stub_name + ext, pyfile_path)
+
         self.extensions = all_extensions
```

Comment:
I have an alternate implementation of this logic at https://github.com/timmoon10/TransformerEngine/commit/eea29fb5f65c3dd6dd6a1522206d41f4f1cd7c91.
- It no longer assumes that Paddle is the first non-CMake extension. This is important when building multiple frameworks, e.g. for Pip wheels.
- It keeps the logic for `super().run()` separate from the Paddle stuff.

Reply:
Done cherry-picking https://github.com/timmoon10/TransformerEngine/commit/eea29fb5f65c3dd6dd6a1522206d41f4f1cd7c91